src/vs/editor/contrib/linesOperations/linesOperations.ts

```diff
@@ -112,18 +112,74 @@ abstract class AbstractMoveLinesAction extends EditorAction {
 		let commands: ICommand[] = [];
 		let selections = editor.getSelections() || [];
 		let autoIndent = editor.getConfiguration().autoIndent;
+		let movingMultipleLines = false;
+		let selectionDirectionDown = true;
+		let newSelections: Selection[] = [];
+
+		if (selections.length > 1) {
+			movingMultipleLines = selections[0].endLineNumber !== selections[selections.length - 1].endLineNumber;
+
+			if (movingMultipleLines) {
+				// Stash selections while processing, set new selections +/- line-change only
+				for (const selection of selections) {
+					let startCol = selection.getStartPosition().column;
+					let startLine = selection.getStartPosition().lineNumber;
+					let newStartLine = this.down ? startLine + 1 : startLine - 1;
+
+					let endCol = selection.getEndPosition().column;
+					let endLine = selection.getEndPosition().lineNumber;
+					let newEndLine = this.down ? endLine + 1 : endLine - 1;
+
+					newSelections.push(new Selection(newStartLine, startCol, newEndLine, endCol));
+				}
+
+				// Work only with one selection per line
+				selections = selections.filter((s, idx, arr) => {
+					return arr.map(sel => sel['endLineNumber']).indexOf(s['endLineNumber']) === idx;
+				});
+
+				selectionDirectionDown = selections[0].endLineNumber < selections[selections.length - 1].endLineNumber;
```

Comment:
I am not sure I understand this piece of code. Why do we calculate the selectionDirectionDown. What does it mean?

Reply:
Yes, if I do this:
![image](https://github.com/user-attachments/assets/3163d7d7-0fb5-48be-b5e1-a7ad5a3e7096)

And select some lines with `Add Previous Occurrence`:
![image](https://github.com/user-attachments/assets/7d77fbc2-ff80-4b0e-930a-77a75dac3917)

And move down:
![image](https://github.com/user-attachments/assets/15d8076c-a8eb-4081-a495-dae982d59358)


vs

![image](https://github.com/user-attachments/assets/c418d4b8-82a9-4e59-ac2b-b2603982bf06)

`Add Previous Occurrence`:
![image](https://github.com/user-attachments/assets/7d77fbc2-ff80-4b0e-930a-77a75dac3917)

move down:
![image](https://github.com/user-attachments/assets/a46f698c-eb1a-46a8-9bee-3adab014d86c)

Reply:
It's because "line 3" gets processed first -> moved to row 6, then the next selection is processed on row 6, which is now "line 3" -> moved to row 7, next selection gets processed on row 7, which is "line 3" -> ends up on row 8. So this is why process order must be right :)